### PR TITLE
Add tests for ParentNode#replaceChildren

### PR DESCRIPTION
**Pull request by saschanaz**
Relates to https://github.com/whatwg/dom/issues/478 and https://github.com/whatwg/dom/pull/851.

Closes #21810

### CODE REVIEW

**Comment by annevk** [comment]
This looks like a good start. We should probably have some tests covering mutation observers as well. And maybe with nodes moving between documents.

**Comment by saschanaz** [comment]
>We should probably have some tests covering mutation observers as well. And maybe with nodes moving between documents.

Should I add same tests for `.append` and `.prepend` too?

**Comment by annevk** [comment]
Yeah, that might be good if they don't have any. Also, feel free to use `const` wherever possible. Bit easier to read than `var`.

**Comment by annevk** [comment]
Are you planning on testing mutation observers doing the right thing in this PR? Or do that separately?

**Comment by saschanaz** [comment]
> 
> 
> Are you planning on testing mutation observers doing the right thing in this PR? Or do that separately?

Oops, I forgot that 🤦‍♀️ I'll try that here.

File: dom/nodes/ParentNode-replaceChildren.html [hunk 1]
```diff
@@ -0,0 +1,135 @@
+<!DOCTYPE html>
+<meta charset=utf-8>
+<title>ParentNode.replaceChildren</title>
+<link rel=help href="https://dom.spec.whatwg.org/#dom-parentnode-replacechildren">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="pre-insertion-validation-hierarchy.js"></script>
+<script>
+  Node.prototype.replaceChildren = function (...args) {
```
Comment (annevk) [inline dom/nodes/ParentNode-replaceChildren.html:9]: I suspect you meant to remove this, right?

**Review by annevk — COMMENTED** [review]
@smaug---- can you take a second look?

File: dom/nodes/pre-insertion-validation-hierarchy.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+ * Validations where `child` argument is irrelevant.
+ * @param {Function} methodName
+ */
+function preInsertionValidateHierarchy(methodName) {
+  function insert(parent, node) {
+    if (parent[methodName].length > 1) {
+      parent[methodName](node, null);
+    } else {
+      parent[methodName](node);
+    }
```
Comment (smaug----) [inline dom/nodes/pre-insertion-validation-hierarchy.js:11]: Is this stuff needed?

**Review by smaug---- — CHANGES_REQUESTED** [review]

File: dom/nodes/pre-insertion-validation-hierarchy.js (hunk 2 above)
Comment (saschanaz) [reply dom/nodes/pre-insertion-validation-hierarchy.js:11]: Yes as `insertBefore` needs a second argument. I can't blindly pass `null` as it causes moving `node` into a `DocumentFragment`.

**Review by smaug---- — APPROVED** [review]

**Pull request merged**